src/opm/input/eclipse/Schedule/KeywordHandlers.cpp

```diff
@@ -564,27 +564,69 @@ File {} line {}.)", wname, location.keyword, location.filename, location.lineno)
+                            if (!apply_default_liquid_target)
+                                 production.production_controls += static_cast<int>(Group::ProductionCMode::LRAT);
+                        }
+                    }
+                    else if (production.cmode == Group::ProductionCMode::WRAT){
                         production.production_controls += static_cast<int>(Group::ProductionCMode::WRAT);
-
-                    if (!apply_default_liquid_target)
+                        if (exceedAction == Group::ExceedAction::RATE) {
+                            if (!apply_default_oil_target)
+                                 production.production_controls += static_cast<int>(Group::ProductionCMode::ORAT);
+                            if (!apply_default_gas_target)
+                                 production.production_controls += static_cast<int>(Group::ProductionCMode::GRAT);
+                            if (!apply_default_liquid_target)
+                                 production.production_controls += static_cast<int>(Group::ProductionCMode::LRAT);
+                        }
+                    }
+                    else if (production.cmode == Group::ProductionCMode::LRAT){
                         production.production_controls += static_cast<int>(Group::ProductionCMode::LRAT);
+                        if (exceedAction == Group::ExceedAction::RATE) {
+                            if (!apply_default_oil_target)
+                                 production.production_controls += static_cast<int>(Group::ProductionCMode::ORAT);
+                            if (!apply_default_gas_target)
+                                 production.production_controls += static_cast<int>(Group::ProductionCMode::GRAT);
+                            if (!apply_default_water_target)
+                                 production.production_controls += static_cast<int>(Group::ProductionCMode::WRAT);
+                        }
+                    }
+                    else {
+                        if (exceedAction == Group::ExceedAction::RATE) {
+                            if (!apply_default_oil_target)
+                                 production.production_controls += static_cast<int>(Group::ProductionCMode::ORAT);
+                            if (!apply_default_water_target)
+                                 production.production_controls += static_cast<int>(Group::ProductionCMode::WRAT);
+                            if (!apply_default_gas_target)
+                                 production.production_controls += static_cast<int>(Group::ProductionCMode::GRAT);
+                            if (!apply_default_liquid_target)
+                                 production.production_controls += static_cast<int>(Group::ProductionCMode::LRAT);
+                        }
+                    }
 
                     if (!apply_default_resv_target)
                         production.production_controls += static_cast<int>(Group::ProductionCMode::RESV);
```

Comment:
Why is the same logic not applied to RESV?

Reply:
According to the commercial simulator description the procedure of exceeding violation of the RESV limit (specified in item 14) is always RATE and thus any setting of item 7 should be ignored to be consistent with the commercial simulator.